pilot/pkg/bootstrap/server.go

```diff
@@ -547,6 +557,44 @@ func (s *Server) createK8sServiceControllers(serviceControllers *aggregate.Contr
 	return
 }
 
+// Check all nodes known to the cluster at the bootstrap time and check for the presence
+// of two labels. If one of nodes is missing at least one label, this function will return true
+func checkNodeAZLabel(s *Server) (bool, error) {
+	nodes, err := s.kubeClient.CoreV1().Nodes().List(meta_v1.ListOptions{})
+	if err != nil {
+		return false, err
+	}
+	for _, n := range nodes.Items {
+		_, exists := n.Labels[NodeRegionLabel]
+		if !exists {
+			return true, nil
+		}
+		_, exists = n.Labels[NodeZoneLabel]
+		if !exists {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func labelClusterNodes(k8s kubernetes.Interface, az string) error {
```

Comment:
I don't like this - too intrusive, and requires write permissions. We should not modify the cluster.

Reply:
labels are used for indication of region/label, and labels needs to be applied somewhere. In the current code, it is expected to see these labels on nodes. IMHO there is no any issue in providing what is expected by the existing code. BTW, this is done ONLY for the case if service registry is kubernetes, so it should not impact other service registries.

If labeling is not acceptable, I am opened for other sugestions, so far that is the only I could see which would be reliable and not depending much on other things.

Reply:
I agree this is too intrusive, we shouldn't require the ability to write to nodes. I think the solution should probably just be to handle the error better on both sides of this communication.